app/src/main/java/com/dd3boh/outertune/playback/DiscordRPCInfoFetch.kt

```diff
@@ -0,0 +1,185 @@
+    rpc.token = discordToken
+    rpc.closeRPC()
+}
+
+@SuppressLint("SetJavaScriptEnabled", "CoroutineCreationDuringComposition")
+@OptIn(ExperimentalMaterial3Api::class, DelicateCoroutinesApi::class)
+fun createDiscordRPC(player: Player, ctx: Context){
+    val discordToken = ctx.dataStore.get(DiscordTokenKey, "")
+    rpc.token = discordToken
+
+    while (rpc.isRpcRunning()){
+        rpc.closeRPC()
+    }
+    val enableRPC = ctx.dataStore.get(EnableDiscordRPCKey, true)
+    val showArtistAvatar = ctx.dataStore.get(ShowArtistRPCKey, true)
+
+    if (discordToken != "" || !enableRPC){
+        val client = HttpClient()
+        val clientDiscordCDN = HttpClient()
+
+        var mediaID = player.currentMediaItem?.mediaId
+        var title = player.currentMediaItem?.mediaMetadata?.title.toString()
+        var album = if (title == player.currentMediaItem?.mediaMetadata?.albumTitle.toString()) "Single" else player.currentMediaItem?.mediaMetadata?.albumTitle.toString()
+        var artist = player.currentMediaItem?.mediaMetadata?.artist.toString()
+        var artwork = player.currentMediaItem?.mediaMetadata?.artworkUri.toString()
+
+        if (title == "null"){
+            rpc.closeRPC()
+            return
+        }
+
+        if (showArtistAvatar){
+            CoroutineScope(Dispatchers.IO).launch {
+            try {
+
+                val response: HttpResponse = client.get("https://pipedapi.r4fo.com/streams/" + mediaID) {
+                }
+                val responseBody: String = response.bodyAsText()
+                
+                var artistArtwork = ""
+                var artistArtworkCDN = ""
+                var artworkCDN = ""
+                
+                val json = JSONObject(responseBody)
+                artistArtwork = json.getString("uploaderAvatar")
+
+                CoroutineScope(Dispatchers.IO).launch {
+                    try {
+                        val response: HttpResponse = clientDiscordCDN.get("https://kizzyapi-1-z9614716.deta.app/image?url=" + artistArtwork) {
+                        }
+                        artistArtworkCDN = JSONObject(response.bodyAsText()).getString("id")
+
+                    } catch (e: Exception) {
+                        println("Error: ${e.message}")
+                        
+                    } finally {
+                        clientDiscordCDN.close()
+                    }
+                }
+                CoroutineScope(Dispatchers.IO).launch {
+                    try {
+                        val response: HttpResponse = clientDiscordCDN.get("https://kizzyapi-1-z9614716.deta.app/image?url=" + artwork) {
+                        }
+
+                        artworkCDN = JSONObject(response.bodyAsText()).getString("id")
+
+                        Thread.sleep(500)
+                        if (artistArtworkCDN == ""){
+                            artistArtworkCDN = "mp:external/_jGArMHI-5rpJu4qVDuiBARu8iEXnHeT0SZS6tZnZug/https/i.imgur.com/zDxXZKk.png"
```

Comment:
Why externally fetch image? Use an icon/image/drawable as a placeholder

Reply:
To my knowledge, Discord RPC needs to use an image from their CDN. A normal drawable/image file will need an extra HTTP request to be uploaded to the CDN. Over here, I am using a pre-uploaded image of a placeholder profile picture so I wouldn't need to make an extra request to the API.